src/js/components/SchemaField.js

```diff
@@ -0,0 +1,315 @@
+
+      if (value === "") {
+        onChange("");
+      } else {
+        onChange(Number(value));
+      }
+    };
+
+    let field = (
+      <FieldInput
+        id={name}
+        type={"number"}
+        className={"field-input-number-narrow"}
+        autoFocus={autofocus}
+        name={name}
+        value={formData}
+        onChange={handleChange.bind(this)}
+        onBlur={onBlur && (event => onBlur(name, event.target.value))}
+        onFocus={onFocus && (event => onFocus(name, event.target.value))}
+      />
+    );
+    if (autofocus) {
+      field = (
+        <FieldAutofocus>
+          {field}
+        </FieldAutofocus>
+      );
+    }
+
+    return (
+      <div>
+        <FieldLabel>
+          {this.getFieldHeading(required, name, schema.description)}
+        </FieldLabel>
+        {field}
+        <FieldError>{errorMessage}</FieldError>
+      </div>
+    );
+  }
+
+  getFieldHeading(required, name = "", description) {
+    let requiredSymbol = null;
+    if (required) {
+      requiredSymbol = (
+        <FormGroupHeadingContent className="text-danger" primary={false}>
+          *
+        </FormGroupHeadingContent>
+      );
+    }
+
+    return (
+      <FormGroupHeading>
+        <FormGroupHeadingContent primary={true}>
+          {name.split(/_|-/).join(" ")}
+        </FormGroupHeadingContent>
+        {requiredSymbol}
+        <FormGroupHeadingContent primary={false}>
+          <Tooltip
+            content={description}
+            interactive={true}
+            maxWidth={300}
+            wrapText={true}
+          >
+            <Icon color="grey" id="circle-question" size="mini" />
+          </Tooltip>
+        </FormGroupHeadingContent>
+      </FormGroupHeading>
+    );
+  }
+
+  getFieldContent(errorMessage, autofocus) {
+    const { schema } = this.props;
+    const RADIO_SELECT_THRESHOLD = 4;
+
+    switch (schema.type) {
+      case "boolean":
+        return this.renderCheckbox(errorMessage, this.props);
+      case "string":
+        if (schema.enum && schema.enum.length <= RADIO_SELECT_THRESHOLD) {
+          return this.renderRadioButtons(errorMessage, this.props);
+        }
+
+        if (schema.enum && schema.enum.length > RADIO_SELECT_THRESHOLD) {
+          return this.renderSelect(errorMessage, autofocus, this.props);
+        }
+
+        return this.renderTextInput(errorMessage, autofocus, this.props);
+      case "integer":
+        return this.renderNumberInput(errorMessage, autofocus, this.props);
+      case "number":
+        return this.renderNumberInput(errorMessage, autofocus, this.props);
+    }
+  }
+
+  render() {
+    const { schema, errorSchema, uiSchema } = this.props;
+
+    if (schema.type === "object") {
+      return <DefaultSchemaField {...this.props} />;
+    }
+
+    const autofocus = uiSchema && uiSchema["ui:autofocus"];
+
+    let errorMessage;
+    if (errorSchema) {
+      errorMessage = errorSchema.__errors.map(error => {
```

Comment:
I don't think there's a better way. The `errorSchema` object is delivered to the schema form onChange handler, and I also parse the `__errors` property in `FrameworkConfigurationForm`.
<img width="465" alt="screen shot 2017-10-26 at 2 03 21 pm" src="https://user-images.githubusercontent.com/1527504/32076925-7f1d8b42-ba56-11e7-989b-b9c8d8fa9e2e.png">

Reply:
I've looked into the `react-jsonschema-form` implementation to understand why they're using a property named `__errors` as it looks a lot like a property meant for _internal_ use only. They're using `__errors` to avoid name collision with a possible sub field named "errors" (for details see: [mozilla-services/react-jsonschema-form/src/validate.js#L56](https://github.com/mozilla-services/react-jsonschema-form/blob/6322579b570336a398dc9faf0870875480455d33/src/validate.js#L56)), but they also provide util to transform `errorSchema` into an error list, so that we don't need to access  `__errors`  directly. Please use [`toErrorList`](https://github.com/mozilla-services/react-jsonschema-form/blob/6322579b570336a398dc9faf0870875480455d33/src/validate.js#L67) instead of directly accessing the property or create a custom util to at least contain the usage as there is a risk that they will change the variable name. I'm ok if you do so in a subsequent PR to no further block this PR.

/cc @dcos/frontend-foundation

Reply:
Interesting. I looked into this a bit, and have created a Jira to prevent blocking this PR any further: https://jira.mesosphere.com/browse/DCOS-19332